### DISCUSSION TITLE
Is SparseMatrix use nativeMKL to speed up in v4.7?

### DISCUSSION THREAD

**Discussion by tmp-coder** [mathnet/mathnet-numerics#821] [Q&A] [answered]
hi, all

i do a test on sparse matrix multiply, (rowCount, colCount) = (1000,1000), and nonzero count = 22328, and sparse matrix is here https://paste.ubuntu.com/p/c82q42yKsf/

while **result** show mathnet.numerics is slower than scipy about 5 times, scipy run 0.007s, while mathnet run 0.036s 

**code**

```c#
var mat = GenerateRandomSparseMatrix(nr, nc, true);
           
 var elapsedtime = Utils.MeasureTime<Matrix<double>>(() => (mat.Transpose().Multiply(mat)));
  Console.WriteLine("elapsed time = {0}s", elapsedtime);
```

py 

```py
mat = pd.read_csv(filePath,header=None).values

mat = sps.csr_matrix(mat)
print("elapsed = {}s",utils.measure_time(lambda : mat.T * mat))
```

**Comment by cdrnet** [comment]
No, not yet unfortunately (only dense).

**Comment by cdrnet** [comment, marked as answer]
There is a sparse solver native provider now - but the other sparse operations are still not covered.
